reactive/filebeat.py

```diff
@@ -193,6 +197,36 @@ def check_filebeat_repo():
     remove_state("filebeat.repo.changed")
 
 
+@when("apt.installed.filebeat")
+@when("config.changed")
```

Comment:
non-blocking: I suspect these `@when` are unnecessary since you're calling this function from within `render_filebeat_template()`

Reply:
`@when("apt.installed.filebeat")` here will cause this function run twice. This should be remove.

Reply:
agreed `@when("apt.installed.filebeat")` is not necessary, but isn't `@when("config.changed")` needed to call the function upon a `config.changed` hook or after changes to the config?  Unless the latter happens because `beat.render` is set upon any config change which would run `render_filebeat_template`, if this is the case then I guess we could live with the sanity checks only running when `render_filebeat_template` is called.

Reply:
`@when("config.changed")` is set in [beat-layers](https://github.com/canonical/layer-beats-base/blob/master/reactive/beats_base.py#L9-L12) and it also set `beat.render` flag on each `config_changed`